portals/knowledge-management/src/Pages/ViewArticle.vue

```diff
@@ -81,13 +91,36 @@
 
         get(props.apiUrl + '/categories/' + route.params.categoryId + '/articles/' + route.params.articleId).then(
             (response) => {
+                if (response.data.guest_id) {
+                    setguestId(response.data.guest_id);
+                }
+
                 category.value = response.data.category;
                 article.value = response.data.article;
                 loading.value = false;
                 portalViewCount.value = response.data.portal_view_count;
+                feedback.value = response.data.article.articleVote
+                    ? response.data.article.articleVote.is_helpful
+                    : null;
+                featureFlag.value = response.data.feature_flag;
             },
         );
     }
+    async function toggleFeedback(type) {
+        if (feedback.value === type) {
+            feedback.value = null; // Unselect if the same button is clicked
+        } else {
+            feedback.value = type; // Select the clicked button
+        }
+        const { post } = consumer();
+        const response = await post(props.apiUrl + '/knowledge_base_article_vote/store', {
+            article_vote: feedback.value,
+            articleId: route.params.articleId,
+            userId: user ? user.id : sessionStorage.getItem('guest_id'),
+            userType: user ? 'Contact' : 'PortalGuest',
```

Comment:
You don't need to pass anything about the user here. That should all be able to be retrieved from either the auth guard or the session in the backend.

Reply:
Code Removed